test/app/model/ChessEngine.spec.ts

```diff
@@ -445,3 +444,187 @@ describe('Return player moves without those that cause his king to check', () =>
+    const engine = new ChessEngine(chessBoard);
+    const bishopSquareFrom: Square = { column: 'C', row: 1 };
+    const bishopSquareTo: Square = { column: 'B', row: 2 };
+    const kingSquareFrom: Square = { column: 'E', row: 8 };
+    const kingSquareTo: Square = { column: 'G', row: 8 };
+    const rookSquareFrom: Square = { column: 'H', row: 8 };
+    const rookSquareTo: Square = { column: 'F', row: 8 };
+
+    engine.move(bishopSquareFrom, bishopSquareTo);
+    expect(engine.move(kingSquareFrom, kingSquareTo)).toIncludeSameMembers([
+      { eventType: 'PieceWasMoved', piece: blackKing, from: kingSquareFrom, to: kingSquareTo },
+      { eventType: 'PieceWasMoved', piece: blackRook, from: rookSquareFrom, to: rookSquareTo },
+    ]);
+  });
+
+  it('When all conditions are fulfilled and square B8 is under attack, then long castling can be done', () => {
+    const boardWithPieces: SquareWithPiece = { G1: whiteBishop, E8: blackKing, A8: blackRook };
+    const chessBoard = new Chessboard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const bishopSquareFrom: Square = { column: 'G', row: 1 };
+    const bishopSquareTo: Square = { column: 'H', row: 2 };
+    const kingSquareFrom: Square = { column: 'E', row: 8 };
+    const kingSquareTo: Square = { column: 'C', row: 8 };
+    const rookSquareFrom: Square = { column: 'A', row: 8 };
+    const rookSquareTo: Square = { column: 'D', row: 8 };
+
+    engine.move(bishopSquareFrom, bishopSquareTo);
+    expect(engine.move(kingSquareFrom, kingSquareTo)).toIncludeSameMembers([
+      { eventType: 'PieceWasMoved', piece: blackKing, from: kingSquareFrom, to: kingSquareTo },
+      { eventType: 'PieceWasMoved', piece: blackRook, from: rookSquareFrom, to: rookSquareTo },
+    ]);
+  });
+
+  it('When all conditions are fulfilled but move starts with the rook, then castling will not be done', () => {
+    const boardWithPieces: SquareWithPiece = { E1: whiteKing, H1: whiteRookH1 };
+    const chessBoard = new Chessboard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const rookSquareFrom: Square = { column: 'H', row: 1 };
+    const rookSquareTo: Square = { column: 'F', row: 1 };
+
+    expect(engine.move(rookSquareFrom, rookSquareTo)).toIncludeSameMembers([
+      { eventType: 'PieceWasMoved', piece: whiteRookH1, from: rookSquareFrom, to: rookSquareTo },
+    ]);
+  });
+
+  it('When king is checked, then castling cannot be done', () => {
+    const boardWithPieces: SquareWithPiece = { E1: whiteKing, H1: whiteRookH1, E8: blackRook };
+    const chessBoard = new Chessboard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const kingSquareFrom: Square = { column: 'E', row: 1 };
+    const shortCastlingSquareTo: Square = { column: 'G', row: 1 };
+    const longCastlingSquareTo: Square = { column: 'C', row: 1 };
+
+    expect(engine.move(kingSquareFrom, shortCastlingSquareTo)).toThrowError('Castling cannot be done when the king is checked.');
```

Comment:
To jest z pewnoscia dobry test case :) Jedyne co nie wiem czy musimy rozrozniac, ze ktos chcial "castling" czy nie. Moze wystarczy, ze po prostu ruch tam nie jest mozliwy? Ale to najwyzej zmienisz w trakcie - jak bedziesz uwazala :)

Reply:
Tylko ten ruch jest równoważny z roszadą, na inne pola można królem normalnie uciekać, wiec trochę dziwnie byłoby, jakby był taki komunikat przy "zwykłym" ruchu.

Reply:
Ja tutaj nie wyrzucałbym błędu.
Postarałbym się dodać do pól z możliwym ruchem, ale bez wyrzucania błędu, bo to nie jest konieczne. A głównie nie chcemy, aby pole to było dostepne w możliwych posunięciach.
https://github.com/nowakprojects/CodersCamp2020.Project.TypeScript.Chess/issues/69

Reply:
Chyba nie rozumiem :(

Reply:
Ja zrobiłem podobnie jak Ty w moim zadaniu, wszystko napisałem wrzuciłem w ifa i wyrzyucałem błąd.
No i mi napisał Mati, ze to nie o to chodziło w zadaniu.
No i poprawiłem to tworząc coś takiego jak na obrazku - czyli zwracana była tablica możliwych ruchów.

Ja dodałem tutaj to... i mogłabyś rozwinąć to co jest zwracane o te pola na roszadę
(Ale to może jeszcze @nowakprojects napisze, czy też toi tak widzi.)

![image](https://user-images.githubusercontent.com/41823689/107961411-2d49d600-6fa6-11eb-9622-3bd651eb6ba3.png)